### PR TITLE
Add version flag

### PR DESCRIPTION
**Pull request by rosdyana**
Solve issue #378

### CODE REVIEW

**Comment by sujaypatil96** [comment]
Thank you for clarifying that @rosdyana. I had a couple of other comments though.

File: setup.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 import sys
 from setuptools import setup
 from warnings import warn
+import versioneer
+
+version = versioneer.get_version()
```
Comment (sujaypatil96) [inline setup.py:7]: You're getting the version here, but not passing it to the `setup()`. From the `pandas` repo that you referenced, here is the [line](https://github.com/pandas-dev/pandas/blob/master/setup.py#L649) I'm referring to.

File: setup.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
...
+cmdclass = versioneer.get_cmdclass()
```
Comment (sujaypatil96) [inline setup.py:8]: Similar to `version`, you're not passing `cmdclass` to `setup()` either. Here is the [line](https://github.com/pandas-dev/pandas/blob/master/setup.py#L651) from pandas.

**Review by sujaypatil96 — CHANGES_REQUESTED** [review]
Would you be able to resolve these two comments @rosdyana? I think we should be good to merge it right after.

**Comment by rosdyana** [comment]
@sujaypatil96 Sure, I just pushed a new commit to resolve that issue, thanks!

**Review by sujaypatil96 — APPROVED** [review]
Thank you @rosdyana. I think this PR is good to be merged.🚀

**Comment by sujaypatil96** [comment]
Converting this PR to draft.

Reason: Passing explicit `version` and `cmdclass` options to `setup()` in setup.py without [pbr](https://github.com/openstack/pbr) will cause the application to break. So we need to explore alternative options of implementing this. See this SO [post](https://stackoverflow.com/questions/55921981/how-can-i-use-pbr-version-from-source).

**Comment by sierra-moxon** [comment]
Hi @rosdyana - we just migrated LinkML to use the poetry virtual environment and package management framework.  I think I need to close this PR as it changes files that were removed during this migration.  With that change, have we addressed your versioning concerns?  If not, please do submit another PR and we can address it ASAP.

**Pull request closed without merging**